geonode/resource/manager.py

```diff
@@ -523,12 +544,12 @@ def copy(
                             _maplayer.pk = _maplayer.id = None
                             _maplayer.map = _resource.get_real_instance()
                             _maplayer.save()
-                    to_update = {}
-                    try:
-                        to_update = storage_manager.copy(_resource).copy()
-                    except Exception as e:
-                        logger.exception(e)
 
+                    links = self._copy_data(instance, target=_resource)
+                    files = []
+                    for link in links:
+                        files.extend(link.asset.location)
```

Comment:
cant' we just return the location from the `_copy_data` function instead of the whole object? If yes, this `for` is not needed

Reply:
adding as comment
```python
# we're just merging all the files together: it won't work once we have multiple assets per resource
# TODO: get the files from the proper Asset
```